python-checks/src/main/java/org/sonar/python/checks/BooleanExpressionInExceptCheck.java

```diff
@@ -37,10 +57,62 @@ public void initialize(Context context) {
 
   private static void checkExceptClause(SubscriptionContext ctx) {
     ExceptClause except = (ExceptClause) ctx.syntaxNode();
-    Expression exception = Expressions.removeParentheses(except.exception());
-    if (exception != null && exception.is(Kind.OR, Kind.AND)) {
-      ctx.addIssue(exception, "Rewrite this \"except\" expression as a tuple of exception classes.");
+    Optional.of(except)
+      .map(ExceptClause::exception)
+      .map(Expressions::removeParentheses)
+      .filter(exception -> exception.is(Kind.OR, Kind.AND))
+      .ifPresent(exception -> {
+        var issue = (IssueWithQuickFix) ctx.addIssue(exception, MESSAGE);
+        addQuickFix(issue, exception);
+      });
+  }
+
+  private static List<String> collectNames(Tree expression) {
+    if (expression.is(Kind.OR, Kind.AND)) {
+      var binaryExpression = (BinaryExpression) expression;
+      var leftExceptions = collectNames(binaryExpression.leftOperand());
+      var rightExceptions = collectNames(binaryExpression.rightOperand());
+      var result = new ArrayList<String>();
+      result.addAll(leftExceptions);
+      result.addAll(rightExceptions);
+      return result;
+    } else if (expression.is(Kind.PARENTHESIZED)) {
+      var parenthesizedExpression = (ParenthesizedExpression) expression;
+      return collectNames(parenthesizedExpression.expression());
+    } else if (expression.is(Kind.NAME)) {
+      var name = (Name) expression;
+      return List.of(name.name());
+    } else if (expression.is(Kind.QUALIFIED_EXPR)) {
+      var name = TreeUtils.tokens(expression)
+        .stream()
+        .map(Token::value)
+        .collect(Collectors.joining());
+      return List.of(name);
     }
+    throw new IllegalArgumentException("Unsupported kind of tree element: " + expression.getKind().name());
+  }
+
+  private static void addQuickFix(IssueWithQuickFix issue, Tree expression) {
+    expression = Objects.requireNonNullElse(TreeUtils.firstAncestorOfKind(expression, Kind.PARENTHESIZED), expression);
+
+    List<String> names;
+    try {
+      names = collectNames(expression);
+    } catch (IllegalArgumentException e) {
+      LOG.error("Could not add quick fix", e);
```

Comment:
I think logging an error here is a tad too strong. We actually expect this will happen in cases a call expression is involved, for example.
In such case, we simply don't want to raise a quick fix. I think it's enough to add a comment here saying that the expression contains subexpressions that are out of scope for quick fixing and that we don't provide one, and avoid logging anything.

Reply:
done